src/fsae_drivers/fsae_can.c

```diff
@@ -0,0 +1,241 @@
+        case CAN_BMS_ERROR_OVER_CURRENT:
+            return true;
+        default:
+            return false;
+    }
+}
+void Send_Bms_Heartbeat(BMS_INPUT_T *bms_input, BMS_STATE_T * bms_state) {
+    Can_Bms_Heartbeat_T bmsHeartbeat;
+    Can_Bms_ErrorID_T error_type = get_error_status();
+    if (error_type != CAN_BMS_ERROR_NONE) {
+        if (is_pack_error(error_type)) {
+            bmsHeartbeat.state = CAN_BMS_STATE_BATTERY_FAULT;
+        } else {
+            bmsHeartbeat.state = CAN_BMS_STATE_BMS_FAULT;
+        }
+    } else {
+        switch (bms_state->curr_mode) {
+            case BMS_SSM_MODE_INIT: 
+                bmsHeartbeat.state = CAN_BMS_STATE_INIT;
+                break;
+            case BMS_SSM_MODE_STANDBY:
+                bmsHeartbeat.state = CAN_BMS_STATE_STANDBY;
+                break;
+            case BMS_SSM_MODE_CHARGE:
+                bmsHeartbeat.state = CAN_BMS_STATE_CHARGE;
+                break;
+            case BMS_SSM_MODE_BALANCE:
+                bmsHeartbeat.state = CAN_BMS_STATE_BALANCE;
+                break;
+            case BMS_SSM_MODE_DISCHARGE:
+                bmsHeartbeat.state = CAN_BMS_STATE_DISCHARGE;
+                break;
+            default:
+                // You should never reach here
+                Board_Println("Unexpected curr_mode");
+                Error_Assert(ERROR_CONTROL_FLOW, bms_input->msTicks);
+                break;
+        }
+    }
+
+    // TODO when we get SOC working
+    bmsHeartbeat.soc = 0;
+
+    Can_Bms_Heartbeat_Write(&bmsHeartbeat);
+}
+
+void Send_Bms_Errors(void) {
+    Can_Bms_Error_T error_msg;
+    error_msg.type = get_error_status();
+    Can_Bms_Error_Write(&error_msg);
+}
+
+Can_Bms_ErrorID_T bms_error_to_can_error(ERROR_T error) {
+    switch (error) {
+        case ERROR_LTC6804_PEC:
+            return CAN_BMS_ERROR_LTC6804_PEC;
+        case ERROR_LTC6804_CVST:
+            return CAN_BMS_ERROR_LTC6804_CVST;
+        case ERROR_LTC6804_OWT:
+            return CAN_BMS_ERROR_LTC6804_OWT;
+        case ERROR_EEPROM:
+            return CAN_BMS_ERROR_EEPROM;
+        case ERROR_CELL_UNDER_VOLTAGE:
+            return CAN_BMS_ERROR_CELL_UNDER_VOLTAGE;
+        case ERROR_CELL_OVER_VOLTAGE:
+            return CAN_BMS_ERROR_CELL_OVER_VOLTAGE;
+        case ERROR_CELL_UNDER_TEMP:
+            return CAN_BMS_ERROR_CELL_UNDER_TEMP;
+        case ERROR_CELL_OVER_TEMP:
+            return CAN_BMS_ERROR_CELL_OVER_TEMP;
+        case ERROR_OVER_CURRENT:
+            return CAN_BMS_ERROR_OVER_CURRENT;
+        case ERROR_CAN:
+            return CAN_BMS_ERROR_CAN;
+        case ERROR_CONFLICTING_MODE_REQUESTS:
+            return CAN_BMS_ERROR_CONFLICTING_MODE_REQUESTS;
+        case ERROR_VCU_DEAD:
+            return CAN_BMS_ERROR_VCU_DEAD;
+        case ERROR_CONTROL_FLOW:
+            return CAN_BMS_ERROR_CONTROL_FLOW;
+        case ERROR_BRUSA:
+        case ERROR_NUM_ERRORS:
+            return CAN_BMS_ERROR_OTHER;
```

Comment:
Oh whoops again, missed that you have a TODO for this already. I'll leave that one for you then.